docs/docs/guides/web3_plugin_guide/plugin_users.md

```diff
@@ -0,0 +1,177 @@
+
+Below are a couple examples of registering the following `SimplePlugin` with various classes imported from various Web3.js packages:
+
+This is an example plugin being used for demonstration purposes:
+
+```typescript
+import { Web3PluginBase } from 'web3-core';
+
+export class SimplePlugin extends Web3PluginBase {
+	public pluginNamespace = 'simplePlugin';
+
+	public simpleMethod() {
+		return 'simpleValue';
+	}
+}
+```
+
+:::caution
+The following code does not include the [module augmentation](/docs/guides/web3_plugin_guide/#module-augmentation) necessary to provide type safety and hinting when using a registered plugin, please refer to the [Setting Up Module Augmentation](/docs/guides/web3_plugin_guide/plugin_users#setting-up-module-augmentation) section for how to augment the `Web3` module to enable typing features for a plugin.
+:::
+
+```typescript
+import Web3 from 'web3';
+import SimplePlugin from 'web3-plugin';
+
+const web3 = new Web3('http://127.0.0.1:8545');
+web3.registerPlugin(new SimplePlugin());
+
+// @ts-expect-error Property 'simplePlugin' does not exist on type 'Web3'
+web3.simplePlugin.simpleMethod();
+```
+
+#### Why is `@ts-expect-error` Required
+
+`// @ts-expect-error Property 'simplePlugin' does not exist on type 'Web3'` is required in order for TypeScript to compile this code; This is due to the lack of module augmentation. `.simplePlugin` is not apart of the standard interface of the `Web3` class that TypeScript is made aware of when we import `Web3`. Module augmentation is us telling TypeScript,
+
+_"Hey, we're modifying the `Web3` class interface to include the interface of `SimplePlugin`"_
+
+and TypeScript will be able to infer the interface of `SimplePlugin` so that we no longer receive an error when calling `web3.simplePlugin.simpleMethod();`
+
+## Setting Up Module Augmentation
+
+This section of the guide will delve deeper into setting up module augmentation, if you run into any issues, please don't hesitate to [create an issue](https://github.com/web3/web3.js/issues/new/choose) or drop a message in the `web3js-general` channel in the ChainSafe [Discord](https://discord.gg/yjyvFRP), and someone from the team/community will assist you.
+
+### Creating a Export Helper File
+
+There exists a [limitation](https://github.com/web3/web3.js/pull/5393/#discussion_r1000727269) with TypeScript's module augmentation: it can only handle _named modules_. So that Web3.js stays backwards compatible, our most commonly used modules (e.g. `Web3`, `Web3Eth`, `Contract`) are exported as `default` exports and are not explicitly named as required by TypeScript for module augmentation. The workaround for this issue is to create a seperate file within your project where you import the default module you wish to augment and re-export it as a named module:
```

Comment:
```suggestion
There exists a [limitation](https://github.com/web3/web3.js/pull/5393/#discussion_r1000727269) with TypeScript's module augmentation: it can only handle _named modules_. So that Web3.js stays backwards compatible, our most commonly used modules (e.g. `Web3`, `Web3Eth`, `Contract`) are exported as `default` exports and are not explicitly named as required by TypeScript for module augmentation. The workaround for this issue is to create a separate file within your project where you import the default module you wish to augment and re-export it as a named module:
```